### PR TITLE
docs: Fixed typos in `semi.md`

### PR DESCRIPTION
**Pull request by alundiak**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

File: docs/src/rules/semi.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ This rule has two options, a string option and an object option.
 String option:
 
 * `"always"` (default) requires semicolons at the end of statements
-* `"never"` disallows semicolons as the end of statements (except to disambiguate statements beginning with `[`, `(`, `/`, `+`, or `-`)
+* `"never"` disallows semicolons at the end of statements (except to disambiguate statements beginning with `[`, `(`, `/`, `+`, or `-`)
```
Comment (mdjermanovic) [inline docs/src/rules/semi.md:79]: I'm not sure if this was a typo. With "as", this means that the rule disallows semicolons that represent the end of a statement, which sounds right. Curious what others think about this.

Comment (alundiak) [reply docs/src/rules/semi.md:79]: I compared previous statement, and for me it resemble the same goal.

Comment (fasttime) [reply docs/src/rules/semi.md:79]: To me, both *as* and *at* sound okay here. Considering that "at the end" is used in the rest of the file without exceptions, I tend to believe that this was indeed an oversight.

Comment (mdjermanovic) [reply docs/src/rules/semi.md:79]: Okay, let's switch to "at" for consistency.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**